src/rusoto/mod.rs

```diff
@@ -339,14 +341,49 @@ impl From<Option<SignedRequestPayload>> for RusotoBody {
     }
 }
 
+static RETRIABLE_CODES: OnceCell<RegexSet> = OnceCell::new();
+
 pub fn is_retriable_error<T>(error: &RusotoError<T>) -> bool {
     match error {
         RusotoError::HttpDispatch(_) => true,
-        RusotoError::Unknown(res)
-            if res.status.is_server_error()
-                || res.status == http::StatusCode::TOO_MANY_REQUESTS =>
-        {
-            true
+        RusotoError::Unknown(response) => {
+            // This header is a direct indication that we should retry the request. Eventually it'd
+            // be nice to actually schedule the retry after the given delay, but for now we just
+            // check that it contains a positive value.
+            if response
```

Comment:
This could be reworded as an `if` … `else if` … `else` block and avoid the returns.

Reply:
Good idea. I actually went one step further and just expressed the conditional directly.